### PR TITLE
Update pre-commit and fix new issues

### PR DESCRIPTION
**Pull request by Carreau**
#4908 rebased + fixes 

updates:

- [github.com/astral-sh/ruff-pre-commit: v0.3.5 → v0.5.0](https://github.com/astral-sh/ruff-pre-commit/compare/v0.3.5...v0.5.0)
- [github.com/pre-commit/mirrors-clang-format: v18.1.3 → v18.1.8](https://github.com/pre-commit/mirrors-clang-format/compare/v18.1.3...v18.1.8)
- [github.com/codespell-project/codespell: v2.2.6 → v2.3.0](https://github.com/codespell-project/codespell/compare/v2.2.6...v2.3.0)
- [github.com/pre-commit/mirrors-mypy: v1.10.0 → v1.10.1](https://github.com/pre-commit/mirrors-mypy/compare/v1.10.0...v1.10.1)

I was not able to make the noqa: U031 (should not use % formatting) to
work, so I went for f-string and doubled the existings brackets. It's
ugly but I don't know what else to do.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -45,9 +45,15 @@ repos:
...
-    rev: "v2.2.6"
+    rev: "v2.3.0"
     hooks:
       - id: codespell
+        args:
+          [
+            "--ignore-words-list",
+            "ags,aray,asend,ba,crate,falsy,feld,inflight,lits,nd,slowy,te,oint,conveniant,atmost",
+          ]
+        exclude: ^(benchmark/benchmarks/pystone_benchmarks/pystone\.py|src/js/package-lock\.json)$
```
Comment (ryanking13) [inline .pre-commit-config.yaml:56]: We added [codespell_ignore_words.txt](https://github.com/pyodide/pyodide/blob/main/tools/codespell_ignore_words.txt) file to manage this list (https://github.com/pyodide/pyodide/pull/4938). So this change needs to be reverted.

**Review by ryanking13 — APPROVED** [review]
Thanks!

File: src/tests/test_pyodide.py [hunk 2]
```diff
@@ -1757,22 +1757,21 @@ def helper(selenium):
 )
 def test_runpython_filename(selenium, run_python):
     msg = selenium.run_js(
-        """
-        try {
-            %s(`
+        f"""
+        try {{
+            {run_python}(`
```
Comment (hoodmane) [inline src/tests/test_pyodide.py:1762]: Can we revert these changes to `test_pyodide.py` and `test_streams.py`? I'd rather use `%s` than have to double all the curly braces. f strings aren't that good for templating JS.

Comment (Carreau) [reply src/tests/test_pyodide.py:1762]: Ok, but I'm not sure how to ignore those with ruff.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Carreau) [reply .pre-commit-config.yaml:56]: ah, sorry I likely messed up the rebase.

File: src/tests/test_pyodide.py (hunk 2 above)
Comment (hoodmane) [reply src/tests/test_pyodide.py]: Okay I disabled UP038 in pyproject.toml

Comment (Carreau) [reply src/tests/test_pyodide.py]: Ah, thanks !
